.github/workflows/linux.yml

```diff
@@ -1,51 +1,46 @@
 name: Linux builds
 on: push
-jobs:
+env:
+  MAX_WARNINGS_GCC_6_Debug:     378
+  MAX_WARNINGS_GCC_6_Release:   661
+  MAX_WARNINGS_GCC_7_Debug:     380
+  MAX_WARNINGS_GCC_7_Release:   329
+  MAX_WARNINGS_GCC_8_Debug:     380
+  MAX_WARNINGS_GCC_8_Release:   330
+  MAX_WARNINGS_GCC_9_Debug:     403
+  MAX_WARNINGS_GCC_9_Release:   358
+  MAX_WARNINGS_Clang_7_Debug:   193
+  MAX_WARNINGS_Clang_7_Release: 193
+  MAX_WARNINGS_Clang_8_Debug:   193
+  MAX_WARNINGS_Clang_8_Release: 193
```

Comment:
I like splitting warnings per different builds a lot… I am a bit afraid it will be hard to maintain.

I just don't understand why Linux debug and release builds have different number of warnigs…

Reply:
It's very troubling how inconsistent the warnings are.. even across architectures as well.. (even through common code without architecture-specific preprocessor branches).  

I think the biggest risk will be introducing new warnings but not being able to isolate them, but in practice merges from upstream are limited to a couple files, and changes we originate here are also usually in a couple files at a time, so hopefully it won't be too bad. 

I was thinking of adding a couple scripts to mass update these warnings too.. like maybe one to switch them to -1s, for open discovery of warning counts. 

 If we can export all of our build logs (named after the compiler, version, and build-type), a script could feed them through the count warnings scripts and then update the yamls with new values.

Id it does become a pain we can incrementally fallback to more and more generalizations.  Like, just gcc vs clang.  or by compiler and build, but ignore version.